homeassistant/components/verisure/binary_sensor.py

```diff
@@ -131,3 +136,53 @@ def available(self) -> bool:
     def device_class(self) -> str:
         """Return the class of this entity."""
         return DEVICE_CLASS_CONNECTIVITY
+
+
+class VerisureFirmwareUpdate(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a Verisure firmware."""
+
+    coordinator: VerisureDataUpdateCoordinator
+
+    @property
+    def name(self) -> str:
+        """Return the name of the binary sensor."""
+        return "Verisure Firmware update"
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the sensor."""
+        return self.coordinator.data["firmware"]["upgradeable"]
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return super().available and self.coordinator.data["firmware"] is not None
```

Comment:
`data["firmware"]` cannot be `None`, it contains a dictionary defined by us.

Reply:
I just dropped `available` override